content-test/addon/Feeds/TopSitesFeed.test.js

```diff
@@ -28,6 +28,40 @@ describe("TopSitesFeed", () => {
     assert.equal(time.minutes(), 15);
   });
 
+  describe("#shouldGetScreenshot", () => {
+    let testSite;
+
+    beforeEach(() => {
+      testSite = {
+        hasMetadata: true,
+        url: "http://www.example.com/",
+        favicon_width: TopSitesFeed.MIN_ICON_SIZE,
+        favicon_height: TopSitesFeed.MIN_ICON_SIZE
+      };
+    });
+    it("should not get a screenshot for sites missing metadata", () => {
+      testSite.hasMetadata = false;
+      assert.equal(false, instance.shouldGetScreenshot(testSite));
```

Comment:
nit: `testSite` already will return `false` as is (as seen below), so changing `hasMetadata` to false doesn't really prove that the condition is working. So I think changing more about the `testSite` (url or favicon dimensions) will make it a better test